public/components/report_definitions/report_settings/report_settings.tsx

```diff
@@ -394,31 +399,122 @@ export function ReportSettings(props) {
     return dashboard_options;
   };
 
-  const getDashboardBaseUrl = () => {
+  const getDashboardBaseUrlCreate = () => {
     let baseUrl = window.location.href;
+    if (edit) {
+      return baseUrl.replace(
+        `opendistro_kibana_reports#/edit/${editDefinitionId}`,
+        'dashboards#/view/'
+      );
+    }
     return baseUrl.replace(
       'opendistro_kibana_reports#/create',
       'dashboards#/view/'
     );
   };
 
-  useEffect(() => {
-    httpClientProps
+  const setReportSourceDropdownOption = (options, report_source, url) => {
+    let index = 0;
+    if (report_source === 'Dashboard') {
+      for (index = 0; index < options.dashboard.length; ++index) {
+        if (url.includes(options.dashboard[index].value)) {
+          setDashboardSourceSelect(options.dashboard[index].value);
+        }
+      }
+    } else if (report_source === 'Visualization') {
+      // todo: add logic once get Visualizations + search is merged
+    } else if (report_source === 'Saved search') {
+      // todo: add logic once get Visualizations + search is merged
+    }
+  };
+
+  const setDefaultFileFormat = (fileFormat) => {
+    let index = 0;
+    for (index = 0; index < PDF_PNG_FILE_FORMAT_OPTIONS.length; ++index) {
+      if (
+        fileFormat.toUpperCase() === PDF_PNG_FILE_FORMAT_OPTIONS[index].label
+      ) {
+        setFileFormat(PDF_PNG_FILE_FORMAT_OPTIONS[index].id);
+      }
+    }
+  };
+
+  const setDefaultEditValues = async (response, reportSourceOptions) => {
+    setReportName(response.report_name);
+    setReportDescription(response.description);
+    reportDefinitionRequest.report_name = response.report_name;
+    reportDefinitionRequest.description = response.description;
+    reportDefinitionRequest.report_params = response.report_params;
+    REPORT_SOURCE_RADIOS.map((radio) => {
```

Comment:
Should this be `forEach` instead of `map` since you are not constructing a new array?

Reply:
Yes, good catch - will change

Reply:
After further investigation, looks like React docs recommend using `map` https://medium.com/javascript-in-plain-english/how-to-loop-through-arrays-in-react-3eaa8a14445